pkg/services/login/authinfo.go

```diff
@@ -107,6 +111,32 @@ func IsExternallySynced(cfg *setting.Cfg, autoProviderLabel string) bool {
 	return true
 }
 
+func IsProviderEnabled(cfg *setting.Cfg, authProviderLabel string) bool {
+	switch authProviderLabel {
+	case SAMLLabel:
+		return cfg.SAMLAuthEnabled
+	case LDAPLabel:
+		return cfg.LDAPAuthEnabled
+	case JWTLabel:
+		return cfg.JWTAuthEnabled
+	case GoogleLabel:
+		return cfg.GoogleAuthEnabled
+	case OktaLabel:
+		return cfg.OktaAuthEnabled
+	case AzureADLabel:
+		return cfg.AzureADEnabled
+	case GitLabLabel:
+		return cfg.GitLabAuthEnabled
+	case GithubLabel:
+		return cfg.GitHubAuthEnabled
+	case GrafanaComLabel:
+		return cfg.GrafanaComAuthEnabled
+	case GenericOAuthLabel:
+		return cfg.GenericOAuthAuthEnabled
```

Comment:
This setting is never read in `readGenericOAuthSettings` 

As such Generic OAuth label will always allow changes 



Extra: It should also be used in `pkg/login/social/social.go` when configuring the Generic oauth to prevent changing config key names from presenting an inaccurate system state. But this can be done later when social is refactored

Reply:
this one is interesting, so they can configure and change the settings. we should make sure to capture that. I guess one thing we need to do as well is to take the env variable in case they spin it up with docker.

Il add that